cl_dll/hud_spectator.cpp

```diff
@@ -1182,7 +1254,7 @@ void CHudSpectator::SetModes(int iNewMainMode, int iNewInsetMode)
 		{
 			char cmdstring[32];
 			// forward command to server
-			sprintf(cmdstring,"specmode %i",iNewMainMode );
+			sprintf(cmdstring,"spec_mode %i",iNewMainMode );
```

Comment:
This will break SetModes for players that use OpenAG DLL on a Half-Life client. I would add a gamedir check to use `spec_mode` in AG and `specmode` in HL and other mods.
```cpp
if (!strcmp(gEngfuncs.pfnGetGameDirectory(), "ag"))
  sprintf(cmdstring,"spec_mode %i",iNewMainMode );
else
  sprintf(cmdstring,"specmode %i",iNewMainMode );
```

Reply:
Alright, then this check has to also be added for the `FindPlayer` function, where I replaced "follow" with the "follownext" stuff.

Reply:
Although, specifically, for the "spec mode" command wouldn't it be easier to send both specmode and spec_mode and let the server decide the "correct answer"?

Reply:
I thought about it but it will spam "Unknown command" messages in the console

Reply:
Which it already did to OpenAG client players on AG servers. :smile: Anyway, ok, will check for the mod dir each time, thanks.